### ISSUE TITLE
Workflow crashes when searching on High Sierra

### ISSUE THREAD

**Issue by yock** [spamwax/alfred-pinboard#23]
The workflow is failing silently on my install of High Sierra. When I open the workflow directory in a terminal and run `./go-pinboard showbookmarks` there (with and without an argument) I get the following stack trace:

```
failed MSpanList_Insert 0x6bc000 0x602f1ce35aff 0x0
fatal error: MSpanList_Insert

runtime stack:
runtime.throw(0x5c272b)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/panic.go:491 +0xad fp=0x7ffeefbff010 sp=0x7ffeefbfefe0
runtime.MSpanList_Insert(0x5e9fe8, 0x6bc000)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/mheap.c:692 +0x8f fp=0x7ffeefbff038 sp=0x7ffeefbff010
MHeap_FreeSpanLocked(0x5e6be0, 0x6bc000, 0x100)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/mheap.c:583 +0x163 fp=0x7ffeefbff078 sp=0x7ffeefbff038
MHeap_Grow(0x5e6be0, 0x8, 0x0)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/mheap.c:420 +0x1a8 fp=0x7ffeefbff0b8 sp=0x7ffeefbff078
MHeap_AllocSpanLocked(0x5e6be0, 0x1, 0x0)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/mheap.c:298 +0x365 fp=0x7ffeefbff0f8 sp=0x7ffeefbff0b8
mheap_alloc(0x5e6be0, 0x1, 0x12, 0x0)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/mheap.c:190 +0x121 fp=0x7ffeefbff120 sp=0x7ffeefbff0f8
runtime.MHeap_Alloc(0x5e6be0, 0x1, 0x10000000012, 0x18849)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/mheap.c:240 +0x66 fp=0x7ffeefbff158 sp=0x7ffeefbff120
MCentral_Grow(0x5ee958, 0x0)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/mcentral.c:197 +0x8b fp=0x7ffeefbff1c0 sp=0x7ffeefbff158
runtime.MCentral_CacheSpan(0x5ee958, 0x0)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/mcentral.c:85 +0x167 fp=0x7ffeefbff1f8 sp=0x7ffeefbff1c0
runtime.MCache_Refill(0x6b8000, 0x12, 0x0)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/mcache.c:90 +0xa0 fp=0x7ffeefbff220 sp=0x7ffeefbff1f8
runtime.mcacheRefill_m()
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/malloc.c:368 +0x57 fp=0x7ffeefbff240 sp=0x7ffeefbff220
runtime.onM(0x4af150)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/asm_amd64.s:273 +0x9a fp=0x7ffeefbff248 sp=0x7ffeefbff240
runtime.mallocgc(0x120, 0x3dba80, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/malloc.go:178 +0x849 fp=0x7ffeefbff2f8 sp=0x7ffeefbff248
runtime.newobject(0x3dba80, 0x6b8000)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/malloc.go:353 +0x49 fp=0x7ffeefbff320 sp=0x7ffeefbff2f8
runtime.newG(0x31b6a)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/proc.go:233 +0x2a fp=0x7ffeefbff338 sp=0x7ffeefbff320
allocg(0x5d70a0)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/proc.c:925 +0x1f fp=0x7ffeefbff348 sp=0x7ffeefbff338
runtime.malg(0x8000, 0x5d7420)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/proc.c:2106 +0x1f fp=0x7ffeefbff378 sp=0x7ffeefbff348
runtime.mpreinit(0x5d7a40)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/os_darwin.c:137 +0x27 fp=0x7ffeefbff390 sp=0x7ffeefbff378
mcommoninit(0x5d7a40)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/proc.c:201 +0xc9 fp=0x7ffeefbff3b8 sp=0x7ffeefbff390
runtime.schedinit()
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/proc.c:138 +0x55 fp=0x7ffeefbff3e0 sp=0x7ffeefbff3b8
runtime.rt0_go(0x7ffeefbff410, 0x3, 0x7ffeefbff410, 0x0, 0x3, 0x7ffeefbff640, 0x7ffeefbff64e, 0x7ffeefbff65c, 0x0, 0x7ffeefbff65e, ...)
	/usr/local/Cellar/go/1.4.1/libexec/src/runtime/asm_amd64.s:95 +0x116 fp=0x7ffeefbff3e8 sp=0x7ffeefbff3e0
```

**Comment by spamwax** [comment]
Not a pro in debugging in stack dumps but my first guess is the issue is not related bug in code but rather some incompatibility between binary produced by go-1.4 and High Sierra.
I am on Sierra and can't reproduce this.

However, if you have latest go installed (or even 1.5+) you can clone this repo and do a `go update && go build` in Terminal and then run `./go-pinboard showbookmarks` again to see if it panics again or not.

**Comment by yock** [comment]
Thanks. I've attempted to build this locally with Go 1.9 and Go 1.8 (both from Homebrew) and it seems the `update` subcommand has been removed from Go. Unless I misunderstand where that `update` command is coming from.

**Comment by yock** [comment]
I have it working. Some unfamiliarity about Go's basic assumptions around `$GOROOT` and `$GOPATH` were unfamiliar to me and I needed to learn. I was able to build the binary with go 1.9, but I still don't understand the origin of the `go update` command. Here are the steps I had to take:

1. `mkdir -p $HOME/go/src`
2. `cd $HOME/go/src`
3. Clone this repo and change to its directory
4. `go get && go build`
5. Open Alfred's workflow settings, `ctrl+click` the workflow and choose "Open in Terminal"
6. `rm ./go-pinboard`
7. `cp $HOME/go/bin/alfred-workflow ./go-workflow`
